modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsDisplayContext.java

```diff
@@ -93,6 +109,50 @@ public List<DropdownItem> getActionDropdownItems() {
 		).build();
 	}
 
+	public HashMap<String, Object> getAssignUserRolesDataMap(
+		SegmentsEntry segmentsEntry) {
+
+		String[] excludedRoleNames = (String[])_httpServletRequest.getAttribute(
+			SegmentsWebKeys.EXCLUDED_ROLE_NAMES);
+
+		ItemSelector itemSelector =
+			(ItemSelector)_httpServletRequest.getAttribute(
+				SegmentsWebKeys.ITEM_SELECTOR);
+
+		RoleItemSelectorCriterion roleItemSelectorCriterion =
+			new RoleItemSelectorCriterion(RoleConstants.TYPE_SITE);
+
+		roleItemSelectorCriterion.setCheckedRoleIds(segmentsEntry.getRoleIds());
+		roleItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+			new UUIDItemSelectorReturnType());
+		roleItemSelectorCriterion.setExcludedRoleNames(excludedRoleNames);
+
+		return HashMapBuilder.<String, Object>put(
+			"itemSelectorURL",
+			() -> {
+				PortletURL portletURL = itemSelector.getItemSelectorURL(
```

Comment:
@Mikellorza, we can also move the variables inside of the function (excludedRoleNames, itemSelector and roleItemSelectorCriterion since we are not using them anywhere else

Reply:
Done @jkappler .

Thanks for your review and for your feedback :)